lib/internal/stream_base_commons.js

```diff
@@ -143,7 +143,7 @@ function afterWriteDispatched(self, req, err, cb) {
   req.async = !!streamBaseState[kLastWriteWasAsync];
 
   if (err !== 0)
-    return self.destroy(errnoException(err, 'write', req.error), cb);
+    return cb(errnoException(err, 'write', req.error));
```

Comment:
Why is this change needed? Should we need to enable `autoDestroy` also on the other classes that use StreamBase as a result? cc @addaleax

Reply:
It's not needed per se. But since we are correcting the rest of the error flow I thought it might be done her as well. Separate PR?

Reply:
It is needed in order to make the stdio-close test pass as expected